### PR TITLE
enforced immutability on args/opts

### PR DESCRIPTION
**Pull request by mihaipgc**
This pull request enforces immutability on the args (opts) object, to make it hashable ([issue](https://github.com/graphcore-research/pyscf-ipu/issues/20))

### CODE REVIEW

File: nanoDFT/nanoDFT.py [hunk 1]
```diff
@@ -396,6 +397,7 @@ def nanoDFT_options(
     mol_str = args["mol_str"]
     del args["mol_str"]
     args = Namespace(**args)
+    args = namedtuple('immutable',vars(args).keys())(**vars(args)) # make immutable
```
Comment (balancap) [inline nanoDFT/nanoDFT.py:400]: Can we have a bit descriptive typename. Like `DFTOptionsImmutable`? I believe that would help if there is an error raise somewhere in the code using it

File: nanoDFT/nanoDFT.py [hunk 2]
```diff
@@ -421,7 +423,6 @@ def nanoDFT_options(
         import mogli
         import imageio
         import matplotlib.pyplot as plt
-        opts.basis = "6-31G"
```
Comment (balancap) [inline nanoDFT/nanoDFT.py:424]: I would keep it, so we don't change the logic of the code in this refactoring. You can update namedtuple in this way:
```
opts = opts._replace(basic="6-31G")
```

**Review by balancap — COMMENTED** [review]
Thanks @mihaipgc for the fix! Just small comments from my side and we can merge

File: nanoDFT/nanoDFT.py (hunk 1 above)
Comment (mihaipgc) [reply nanoDFT/nanoDFT.py]: Thank you, this looks much better now!

File: nanoDFT/nanoDFT.py (hunk 2 above)
Comment (mihaipgc) [reply nanoDFT/nanoDFT.py:424]: Fixed, thanks!

**Pull request merged**